src/Marten/Events/AsyncProjectionTestingExtensions.cs

```diff
@@ -13,62 +16,76 @@ public static class TestingExtensions
     /// </summary>
-    /// <param name="store"></param>
-    /// <param name="timeout"></param>
     public static async Task WaitForNonStaleProjectionDataAsync(this IDocumentStore store, TimeSpan timeout)
     {
         var databases = await store.Storage.AllDatabases().ConfigureAwait(false);
+        var projectionsCount = store.AsyncProjectionsCount();
 
         if (databases.Count == 1)
         {
-            await WaitForNonStaleProjectionDataAsync(databases.Single(), timeout).ConfigureAwait(false);
+            await WaitForNonStaleProjectionDataAsync(databases.Single(), projectionsCount, timeout).ConfigureAwait(false);
         }
         else
         {
-            var tasks = databases.Select(db => db.WaitForNonStaleProjectionDataAsync(timeout));
+            var tasks = databases.Select(db => db.WaitForNonStaleProjectionDataAsync(projectionsCount, timeout));
             await Task.WhenAll(tasks).ConfigureAwait(false);
         }
     }
 
-
     /// <summary>
     /// Wait for any running async daemon for a specific tenant id or database name to catch up to the latest event sequence at the time
     /// this method is invoked for all projections. This method is meant to aid in automated testing
     /// </summary>
-    /// <param name="store"></param>
     /// <param name="tenantIdOrDatabaseName">Either a tenant id or the name of a database within the system</param>
-    /// <param name="timeout"></param>
     public static async Task WaitForNonStaleProjectionDataAsync(this IDocumentStore store, string tenantIdOrDatabaseName, TimeSpan timeout)
     {
         // Assuming there's only one database in this usage
         var database = await store.Storage.FindOrCreateDatabase(tenantIdOrDatabaseName).ConfigureAwait(false);
 
         if (store.Storage is DefaultTenancy)
-
-        await WaitForNonStaleProjectionDataAsync(database, timeout).ConfigureAwait(false);
+            await WaitForNonStaleProjectionDataAsync(database, store.AsyncProjectionsCount(), timeout).ConfigureAwait(false);
     }
 
-    public static async Task WaitForNonStaleProjectionDataAsync(this IMartenDatabase database, TimeSpan timeout)
+    /// <summary>
+    /// Wait for any running async daemon to catch up to the latest event sequence at the time
+    /// </summary>
+    /// <param name="projectionsCount">Will be awaited till all shards have been started before checking if they've caught up with the sequence number</param>
+    /// <exception cref="TimeoutException"></exception>
+    public static async Task WaitForNonStaleProjectionDataAsync(this IMartenDatabase database, int projectionsCount, TimeSpan timeout)
     {
         using var cancellationSource = new CancellationTokenSource();
         cancellationSource.CancelAfter(timeout);
 
-        var initial = await database.FetchEventStoreStatistics(cancellationSource.Token).ConfigureAwait(false);
-        while (initial.EventSequenceNumber == 0 && !cancellationSource.IsCancellationRequested)
```

Comment:
`initial` was evaluated once and never updated within `while` loop. 
never happened to me that no events were present, but i thought it is still better to correct it

Reply:
@a-shtifanov-laya, I don't believe this is necessary. FetchEventStoreStatistics retrieves the maximum sequence from the event table, eliminating the need for any waiting here.